### ISSUE TITLE
Improve new user experience with extended tutorial

### ISSUE THREAD

**Issue by ngcthao** [bitburner-official/bitburner-src#1504]
Although the end of the tutorial directs users to the documentation, it's quite a lengthy read and I imagine it might make some potential players walk away. The other features are fun and I think it's a shame. I think having a secondary or extended tutorial to further explain all the features interactively would be a good addition to improve new user engagement.

Thank you for your time!

**Comment by d0sboots** [comment]
Huh, and here I was thinking the tutorial we had was already over-long... XD

One thing that is certainly true is that after you are dropped in, it is *very much* an open-world/explore-on-your-own type of game. There aren't really any breadcrumbs to hint you towards any particular mechanics, and you can go a long time without discovering a particular powerful technique. This is both one of the biggest strengths *and* weaknesses of BB, IMO.

**Comment by ngcthao** [comment]
I agree with you there. I love open world games and dislike being forced into tutorials explaining things down to the abcs. I imagine the active community also enjoyed challenge, so I suppose it comes down to the desired target player base.

Steep learning curves are usually a big time sink. Which is great for people looking for a challenge, but the generally accepted expectation of reading up and watching community guides right off the bat does seem like a big dedication (in my opinion).

As for the existing tutorial, I did think some parts could be more streamlined if you wanted it shortened. For example, 
- I don't think it's entirely necessary to explain "scan-analyze 2" after "scan analyze" since it's not used yet
- there a two slides referring the user to the documentation that are only 3-4 slides apart
- it may be more intuitive to explain what a command does before asking the user to enter it

I believe making the tutorial smoother can also bring more focus to the important mechanics of the game and while reducing the mental load on new players.

**Comment by d0sboots** [comment]
> Steep learning curves are usually a big time sink. Which is great for people looking for a challenge, but the generally accepted expectation of reading up and watching community guides right off the bat does seem like a big dedication (in my opinion).

Hmm, it's been a while since I've done the tutorial, but I don't think that's the generally accepted expectation. Rather, the docs are mentioned prominently so that if/when you get stuck, you know where to go looking. They're primarily a reference, not a guide, so there is no expectation that you read all of them - I certainly haven't.

Edit: The steepness of the learning curve is going to vary a *lot* depending on your prior experience, though. I came into this as a professional programmer who already knew JavaScript, so it was just a matter of understanding the game's mechanics. If you start without knowing *anything* about programming, the curve will be *much* steeper. There are lots of people who have learned to code while playing BB, but learning from guides or talking to others is more necessary in that case.

**Comment by ngcthao** [comment]
Aah, I see what you mean. I could definitely be wrong in regard to the expectations. I do still think the tutorial has room for improvement, but if the primary struggle for new players (that are also new to js) is the programming half of the game, then there's not much to be done lest the game turns into a coding course. I appreciate your time responding to this!
